lib/middleman-prismic/preview.rb

```diff
@@ -0,0 +1,22 @@
+module Middleman
+  module Prismic
+    class Preview
+      def initialize(app, options={})
+        @app = app
+      end
+
+      def call(env)
+        req = ::Rack::Request.new(env)
+        if req.path =~ %r(^/preview)
+          token = Shellwords.escape(req.params["token"])
+
+          Kernel.system "middleman prismic --ref #{token}"
```

Comment:
For additional safety, skip the shell entirely:

```ruby
Kernel.system "middleman", "prismic", "--ref", token
```

This way we don't have to worry about `token` containing a shell exploit.

Reply:
oh, wooow! yes!

Reply:
Should you check `$?` after calling `#system`?

Reply:
@mike-burns done in 9d14fda